src/main/kotlin/no/nav/familie/felles/utbetalingsgenerator/Utbetalingsgenerator.kt

```diff
@@ -182,12 +183,39 @@ class Utbetalingsgenerator {
         nyeKjeder: List<ResultatForKjede>,
     ): List<AndelMedPeriodeId> =
         nyeKjeder.flatMap { nyKjede ->
-            nyKjede.beståendeAndeler.map { AndelMedPeriodeId(it) } +
-                nyKjede.nyeAndeler.map {
-                    AndelMedPeriodeId(it, behandlingsinformasjon.behandlingId)
+            nyKjede.beståendeAndeler.map { beståendeAndel ->
+                AndelMedPeriodeId(
+                    andel = beståendeAndel,
+                    nyKildeBehandlingId =
+                        bestemKildeBehandlingIdForBeståendeAndel(
+                            beståendeAndel = beståendeAndel,
+                            opphørsandel = nyKjede.opphørsandel?.first,
+                            inneværendeBehandlingId = behandlingsinformasjon.behandlingId,
+                        ),
+                )
+            } +
+                nyKjede.nyeAndeler.map { nyAndel ->
+                    AndelMedPeriodeId(
+                        andel = nyAndel,
+                        nyKildeBehandlingId = behandlingsinformasjon.behandlingId,
+                    )
                 }
```

Comment:
Ingen stor sak, men syns det er litt lettere å lese hvis man deler det opp slik

```suggestion
            val beståendeAndelerPerioder = nyKjede.beståendeAndeler.map { beståendeAndel ->
                AndelMedPeriodeId(
                    andel = beståendeAndel,
                    nyKildeBehandlingId =
                        bestemKildeBehandlingIdForBeståendeAndel(
                            beståendeAndel = beståendeAndel,
                            opphørsandel = nyKjede.opphørsandel?.first,
                            inneværendeBehandlingId = behandlingsinformasjon.behandlingId,
                        ),
                )
            }
            val nyeAndelerPerioder nyKjede.nyeAndeler.map { nyAndel ->
                    AndelMedPeriodeId(
                        andel = nyAndel,
                        nyKildeBehandlingId = behandlingsinformasjon.behandlingId,
                    )
                }
            return beståendeAndelerPerioder + nyeAndelerPerioder
```